pkg/cmd/server/logs.go

```diff
@@ -42,66 +57,137 @@ var logsCmd = &cobra.Command{
 
 		switch {
-		case fileFlag:
-			return readServerLogFile()
-
+		case localFlag:
+			return readLocalServerLogFile()
 		default:
-			ws, res, err := apiclient.GetWebsocketConn(context.Background(), "/log/server", &activeProfile, &query)
+			return readRemoteServerLogFile(ctx, activeProfile, query)
+		}
+	},
+}
 
-			if err != nil {
-				log.Error(apiclient.HandleErrorResponse(res, err))
+func readRemoteServerLogFile(ctx context.Context, activeProfile config.Profile, query string) error {
+	apiclient, err := apiclient_util.GetApiClient(&activeProfile)
+	if err != nil {
+		return err
+	}
 
-				if activeProfile.Id != "default" {
-					return nil
-				}
+	logFiles, res, err := apiclient.ServerAPI.GetServerLogFiles(ctx).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
 
-				readLogsFile := true
-				form := huh.NewForm(
-					huh.NewGroup(
-						huh.NewConfirm().Description("An error occurred while connecting to the server. Would you like to read from local log file instead?").
-							Value(&readLogsFile),
-					),
-				).WithTheme(views.GetCustomTheme())
-				formErr := form.Run()
-				if formErr != nil {
-					return formErr
-				}
+	if len(logFiles) == 0 {
+		return promptReadingLocalServerLogs("No server log files found.")
+	}
 
-				if readLogsFile {
-					return readServerLogFile()
-				}
-				return nil
+	if fileFlag == "" && len(logFiles) == 1 {
+		fileFlag = logFiles[0]
+	}
 
-			}
+	if fileFlag == "" {
+		selectedFile := selection.GetLogFileFromPrompt(logFiles)
+		if selectedFile == nil {
+			return nil
+		}
 
-			for {
-				_, msg, err := ws.ReadMessage()
-				if err != nil {
-					return nil
-				}
+		fileFlag = *selectedFile
+	}
 
-				fmt.Println(string(msg))
-			}
+	if fileFlag != "" {
+		if query != "" {
+			query += "&"
 		}
-	},
+		query += fmt.Sprintf("file=%s", fileFlag)
+	}
+
+	ws, res, err := apiclient_util.GetWebsocketConn(context.Background(), "/log/server", &activeProfile, &query)
+
+	if err != nil {
+		log.Error(apiclient_util.HandleErrorResponse(res, err))
+
+		if activeProfile.Id != "default" {
+			return nil
+		}
+
+		return promptReadingLocalServerLogs("An error occurred while connecting to the server.")
+	}
+
+	for {
+		_, msg, err := ws.ReadMessage()
+		if err != nil {
+			return nil
+		}
+
+		fmt.Println(string(msg))
+	}
+}
+
+func promptReadingLocalServerLogs(info string) error {
+	readLocalLogsFile := true
+	form := huh.NewForm(
+		huh.NewGroup(
+			huh.NewConfirm().Description(info + " Would you like to read from local log files instead?").
+				Value(&readLocalLogsFile),
+		),
+	).WithTheme(views.GetCustomTheme())
+	formErr := form.Run()
+	if formErr != nil {
+		return formErr
+	}
+
+	if readLocalLogsFile {
+		return readLocalServerLogFile()
+	}
+
+	return nil
 }
 
-func readServerLogFile() error {
-	views.RenderBorderedMessage("Reading from server log file...")
+func readLocalServerLogFile() error {
+	views.RenderBorderedMessage("Reading from local server log file...")
```

Comment:
I would eliminate this box if you and @idagelic agree. Don't think it's necessary anymore.